src/chsh.c

```diff
@@ -498,7 +498,7 @@ int main (int argc, char **argv)
 	 * name, or the name getlogin() returns.
 	 */
 	if (optind < argc) {
-		user = argv[optind];
+		user = xstrndup(argv[optind], USER_ENTRY_MAX_LENGTH);
```

Comment:
Why truncate it silently?  If the input is invalid, we should just fail, no?

Reply:
New approach is applied by direct usage of a new function: is_valid_user_name